src/libraries/Microsoft.PowerFx.Core/Public/Values/CollectionTableValue.cs

```diff
@@ -108,6 +108,41 @@ record = Marshal(item);
             }
         }
 
+        public override async Task<DValue<BooleanValue>> RemoveAsync(IEnumerable<FormulaValue> recordsToRemove, bool all = false)
```

Comment:
>RemoveAsync [](http://example.com/codeflow?start=57&length=11)

Looking at these loops reminds me, we should be taking in a CancellationToken parameter (passed in from caller), 
and call 
cancel.ThrowIfCancellationRequested() inside the loop. 
Derived async impls will pass this cancellation token into their network calls.